scopes/pipelines/builder/builder.graphql.ts

```diff
@@ -1,36 +1,123 @@
-import { Component } from '@teambit/component';
+import { Component, ComponentID } from '@teambit/component';
 import gql from 'graphql-tag';
-
+import { Scope } from '@teambit/legacy/dist/scope';
+import { ArtifactObject } from '@teambit/legacy/dist/consumer/component/sources/artifact-files';
+import isBinaryPath from 'is-binary-path';
 import { BuilderMain } from './builder.main.runtime';
+import { PipelineReport } from './build-pipeline-result-list';
+
+type ArtifactGQLFile = {
+  id: string;
+  name: string;
+  path: string;
+  content?: string;
+  downloadUrl?: string;
+};
+type ArtifactGQLData = ArtifactObject & { files: ArtifactGQLFile[]; componentId: ComponentID };
+type TaskReport = PipelineReport & {
+  artifact: ArtifactGQLData;
+  componentId: ComponentID;
+};
+const FILE_PATH_PARAM_DELIM = '~';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export function builderSchema(builder: BuilderMain) {
+export function builderSchema(builder: BuilderMain, scope: Scope) {
   return {
     typeDefs: gql`
-      type ExtensionDescriptor {
-        # extension ID.
-        id: String
+      type TaskReport {
+        id: String!
+        name: String!
+        description: String
+        startTime: String
+        endTime: String
+        errors: [String!]
+        warnings: [String!]
+        artifact(path: String): Artifact
+      }
 
-        # icon of the extension.
-        icon: String
+      type ArtifactFile {
+        # for GQL caching - same as the path
+        id: String!
+        name: String
+        path: String!
+        # artifact file content (only for text files). Use /api/<component-id>/~aspect/builder/<extension-id>/~<path> to fetch binary file data
+        content: String
+        downloadUrl: String
       }
 
-      type ExtensionArtifact {
-        # descriptor of the artifact's extension
-        extensionDescriptor: ExtensionDescriptor
+      type Artifact {
```

Comment:
Artifact should contain more meta data
See Artifact and Artifact-definition files in the artifact folder in the builder